editor/modules/runner/physics-engine.ts

```diff
@@ -0,0 +1,72 @@
+
+interface Entity {
+    pos: Vec2;
+    next_pos: Vec2;
+    nominal_speed: number;
+    speed: Vec2;
+}
+
+interface Vec2 {
+    x: number;
+    y: number;
+}
+
+export class PhysicsEngine {
+
+    // The list of entities except the player.
+    private _entities: Entity[];
+
+    // The player entity
+    player: Entity;
+
+    set entities(entities: Entity[]) {
+        this._entities = entities;
+    }
+
+    update(dt: number) {
+        // Update speed.
+        for (let entity of this._entities) {
+            let dir =  direction(entity.next_pos, entity.pos);
+            entity.speed.x = entity.nominal_speed * dir.x / dt;
+            entity.speed.y = entity.nominal_speed * dir.y / dt;
+        }
+
+        // Update positions
+        let diff;
+        for (let entity of this._entities) {
+
+            diff = entity.next_pos.x - entity.pos.x;
```

Comment:
I assume `next_pos` will hold the updates coming from the server, and will try to do some prediction if non is received, correct?

But after the first iteration, `entity.pos` is assigned to `entity.next_pos`, so this diff will be 0, which I assume is not the intention (it will also affect the direction, thus speed calculated above).

Or perhaps I misunderstood the intent of this code?